tests/scripts/thread-cert/Cert_5_1_01_RouterAttach.py

```diff
@@ -168,6 +179,317 @@ def test(self):
+        #             - Challenge TLV
+        #             - Connectivity TLV
+        #             - Leader Data TLV
+        #             - Link-layer Frame Counter TLV
+        #             - Link Margin TLV
+        #             - Response TLV
+        #             - Source Address
+        #             - Version TLV
+        #             - MLE Frame Counter TLV (optional)
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_dst64(ROUTER).\
+            filter_mle_cmd(MLE_PARENT_RESPONSE).\
+            filter(lambda p: {\
+                              CHALLENGE_TLV,\
+                              CONNECTIVITY_TLV,\
+                              LEADER_DATA_TLV,\
+                              LINK_LAYER_FRAME_COUNTER_TLV,\
+                              LINK_MARGIN_TLV,\
+                              RESPONSE_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV,\
+                              MLE_FRAME_COUNTER_TLV\
+                              } <= set(p.mle.tlv.type) or\
+                             {\
+                              CHALLENGE_TLV,\
+                              CONNECTIVITY_TLV,\
+                              LEADER_DATA_TLV,\
+                              LINK_LAYER_FRAME_COUNTER_TLV,\
+                              LINK_MARGIN_TLV,\
+                              RESPONSE_TLV,\
+                              SOURCE_ADDRESS_TLV,\
+                              VERSION_TLV\
+                               } <= set(p.mle.tlv.type)).\
+                   must_next()
+        end_index = pkts.index
+
+        # verify sub step in Step 2 : Subsequent ones MAY be sent to all Routers and REEDS
+
+        _pkt = pkts_copy.range(start_index, end_index).\
+            filter_wpan_src64(ROUTER).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {\
+                              CHALLENGE_TLV,\
+                              MODE_TLV,\
+                              SCAN_MASK_TLV,\
+                              VERSION_TLV\
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255)\
+            .next()
+
+        if _pkt is not None:
+            _pkt.must_verify(lambda p:
+                    p.mle.tlv.scan_mask.r == 1 and\
+                    p.mle.tlv.scan_mask.e == 1)
+
+        # Step 4: Router sends a MLE Child ID Request.
+        #         The following TLVs MUST be present in the MLE Child ID Request:
+        #             - Link-layer Frame Counter TLV
+        #             - Mode TLV
+        #             - Response TLV
+        #             - Timeout TLV
+        #             - TLV Request TLV
+        #             - Address16 TLV
+        #             - Network Data TLV
+        #             - Route64 TLV (optional)
+        #             - Version TLV
+        #             - MLE Frame Counter TLV (optional)
```

Comment:
If Test Plan says that one TLV is optional, maybe we can just skip checking for it since whether this TLV is included does not impact the verification result. 
But if Test Plan requires extra checking when the TLV field exists, we should add corresponding code like `mle.tlv.xxx is null or mle.tlv.xxx == yyy`.